web/src/api/network.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import { del, get, patch, post, put } from "~/api/http";
+import { APIAccessPoint, APIConnection, APIDevice, NetworkGeneralState } from "~/types/network";
+
+/**
+ * Returns the network configuration
+ */
+const fetchState = (): Promise<NetworkGeneralState> => get("/api/network/state");
+
+/**
+ * Returns a list of known devices
+ */
+const fetchDevices = (): Promise<APIDevice[]> => get("/api/network/devices");
+
+/**
+ * Returns data for given connection name
+ */
+const fetchConnection = (name: string): Promise<APIConnection> =>
+  get(`/api/network/connections/${name}`);
+
+/**
+ * Returns the list of known connections
+ */
+const fetchConnections = (): Promise<APIConnection[]> => get("/api/network/connections");
+
+/**
+ * Returns the list of known access points
+ */
+const fetchAccessPoints = (): Promise<APIAccessPoint[]> => get("/api/network/wifi");
+
+/**
+ * Adds a new connection
+ *
+ * @param connection - connection to be added
+ */
+const addConnection = (connection: APIConnection) => post("/api/network/connections", connection);
+
+/**
+ * Updates given connection
+ *
+ * @param connection - connection to be added
+ */
+const updateConnection = (connection: APIConnection) =>
+  put(`/api/network/connections/${connection.id}`, connection);
+
+/**
+ * Deletes the connection matching given name
+ */
+const deleteConnection = (name: string) => del(`/api/network/connections/${name}`);
+
+/**
+ * Apply network changes
+ */
+const applyChanges = () => post("/api/network/system/apply");
+
+/**
+ * Performs the connect action for connection matching given name
+ */
+const connect = (name: string) => patch(`/api/network/connections/${name}/connect`);
```

Comment:
to be honest, I do not get here why it is patch..for patch it should contain patch structure that changes connection object. https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH mentions that `A PATCH request is considered a set of instructions on how to modify a resource. Contrast this with PUT which is a complete representation of a resource.` So here I expect that for connect and disconnect it is more like action like apply that uses POST above.

Reply:
As commented at https://github.com/openSUSE/agama/pull/1519#discussion_r1714857840, you can think in these actions as the instructions you're requesting for modifying the connection state.

As I said in linked comment, I still having pending to refresh the HTTP verbs/requests. But I tend to think in POST as a request for creating something or appending more data to a resource (although for the latest I'd use a PUT). Also, I expect that servers returns the resource representation or at least where to retrieve it. None of these are the case for connect and disconnect actions for which we are _just_ requiring a change in the connection state, a request _side-effect_.